packages/ion_identity_client/lib/src/signer/password_signer.dart

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'dart:convert';
+
+import 'package:convert/convert.dart';
+import 'package:crypto/crypto.dart';
+import 'package:cryptography/cryptography.dart' as crypto;
+import 'package:ion_identity_client/ion_identity.dart';
+import 'package:ion_identity_client/src/auth/dtos/dtos.dart';
+import 'package:ion_identity_client/src/auth/services/key_service.dart';
+
+class PasswordSigner {
+  PasswordSigner({
+    required this.config,
+    required this.keyService,
+  });
+
+  final KeyService keyService;
+  final IONIdentityConfig config;
+
+  Future<CredentialRequestData> getCredentialInfo({
```

Comment:
The class name is also too specific. From what I see, it signs more than just the `PasswordProtectedKey`.